### PR TITLE
test: Tests that ensure ADR8 non-compliant messages are deadlettered.

### PR DESCRIPTION
**Pull request by defectiveAi**
Added tests that receive non-compliant integration events and verify that these events are placed in the dead-letter queue.

### CODE REVIEW

**Review by lasrinnil — DISMISSED** [review]

File: source/dotnet/IntegrationTests/Fixture/FunctionApp/IntegrationEventListenerFunctionAppFixture.cs [hunk 1]
```diff
@@ -57,6 +62,12 @@ public IntegrationEventListenerFunctionAppFixture()
 
         public TopicResource MarketParticipantChangedTopic { get; private set; } = null!;
 
+        public ServiceBusReceiver MeteringPointCreatedDeadLetter { get; private set; } = null!;
+
+        public ServiceBusReceiver MeteringPointConnectedDeadLetter { get; private set; } = null!;
+
+        public ServiceBusReceiver MarketParticipantChangedDeadLetter { get; private set; } = null!;
+
```
Comment (BjarkeMeier) [inline source/dotnet/IntegrationTests/Fixture/FunctionApp/IntegrationEventListenerFunctionAppFixture.cs:70]: Naming: Is `MeteringPointCreatedDeadLetter` a dead-letter?

File: source/dotnet/IntegrationTests/IntegrationEventListener/MarketParticipantChangedListenerEndpointTests.cs [hunk 2]
```diff
@@ -119,6 +119,34 @@ await FunctionAsserts
             allReceived.Should().BeFalse();
         }
 
+        [Fact]
+        public async Task When_ReceivingMessageWithoutIntegrationEventProperties_Then_MessageIsDeadLettered()
```
Comment (BjarkeMeier) [inline source/dotnet/IntegrationTests/IntegrationEventListener/MarketParticipantChangedListenerEndpointTests.cs:123]: Shouldn't we know exactly which ADR8 props? What if new props are added to ADR8, then it's hard to know whether this test reflects those changes or not.

File: source/dotnet/IntegrationTests/IntegrationEventListener/MeteringPointConnectedListenerEndpointTests.cs [hunk 3]
```diff
@@ -94,6 +94,36 @@ public async Task When_ReceivingMeteringPointConnectedMessage_MeteringPointConne
             actual.EffectiveDate.Should().Be(effectiveDate.ToInstant());
         }
 
+        [Fact]
+        public async Task When_ReceivingMessageWithoutIntegrationEventProperties_Then_MessageIsDeadLettered()
```
Comment (BjarkeMeier) [inline source/dotnet/IntegrationTests/IntegrationEventListener/MeteringPointConnectedListenerEndpointTests.cs:98]: Just a thought: How about making a generic base class holding all tests common to every integration event. That would make it a lot easier to add new events - and also to make sure that all the same tests are applied to all event types.

File: source/dotnet/IntegrationTests/Fixture/FunctionApp/IntegrationEventListenerFunctionAppFixture.cs (hunk 1 above)
Comment (defectiveAi) [reply source/dotnet/IntegrationTests/Fixture/FunctionApp/IntegrationEventListenerFunctionAppFixture.cs:70]: It is a receiver for the deadlettered messages for each topic. Suggestions for better naming are welcome. :)

File: source/dotnet/IntegrationTests/IntegrationEventListener/MarketParticipantChangedListenerEndpointTests.cs (hunk 2 above)
Comment (defectiveAi) [reply source/dotnet/IntegrationTests/IntegrationEventListener/MarketParticipantChangedListenerEndpointTests.cs:123]: The scope of this test is to ensure that functionality which deadletters messages missing an ARD8 property is working correctly.

If we want to ensure that we read all and store all integration event properties, I can add additional test for that. Do note that we are already skipping some properties, like MessageVersion and EventIdentification. Maybe just a test for the properties we use?

File: source/dotnet/IntegrationTests/IntegrationEventListener/MeteringPointConnectedListenerEndpointTests.cs (hunk 3 above)
Comment (defectiveAi) [reply source/dotnet/IntegrationTests/IntegrationEventListener/MeteringPointConnectedListenerEndpointTests.cs:98]: Will do.

File: source/dotnet/IntegrationTests/Fixture/FunctionApp/IntegrationEventListenerFunctionAppFixture.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/IntegrationTests/Fixture/FunctionApp/IntegrationEventListenerFunctionAppFixture.cs:70]: Append `Receiver` to the names...

File: source/dotnet/IntegrationTests/IntegrationEventListener/MarketParticipantChangedListenerEndpointTests.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/dotnet/IntegrationTests/IntegrationEventListener/MarketParticipantChangedListenerEndpointTests.cs:123]: @lasrinnil and I would like a test for each property that is used in the domain.

**Review by johevemi — APPROVED** [review]

**Pull request merged**